src/components/Counter/Counter.jsx

```diff
@@ -1,45 +1,77 @@
-import { useEffect, useState } from 'react';
+import { useEffect, useState, useReducer } from 'react';
 import styles from './Counter.css';
 
+const initialCount = 0;
+
 const colors = {
   yellow: 'rgb(236, 222, 153)',
   green: 'rgb(52, 211, 153)',
   red: 'rgb(239, 68, 68)',
 };
 
+function countReducer(counter, action) {
+  switch (action.type) {
+    case 'increment':
+      return counter + 1;
+    case 'decrement':
+      return counter - 1;
+    case 'reset':
+      return 0;
+    default:
+      return state;
+  }
```

Comment:
Here we will want to throw Error ('action type not recognized') instead of returning the color.